pandas/core/frame.py

```diff
@@ -5444,6 +5444,34 @@ def shift(
     ) -> DataFrame:
         axis = self._get_axis_number(axis)
 
+        # Handle the case of multiple shifts
+        if is_list_like(periods):
+
+            new_df = DataFrame()
+
+            from pandas.core.reshape.concat import concat
+
+            for i in periods:
```

Comment:
Don‘t do this. Collect the frames and concat once at the end.

Reply:
fixed